### DISCUSSION TITLE
Question about Silero VAD implementation and chunking

### DISCUSSION THREAD

**Discussion by ngcheeyuan** [SYSTRAN/faster-whisper#882] [Q&A]
I can't quite find information on how VAD is implemented in faster-whipser. Some details will help.
For example :

Do you all chop up the audio into 30 seconds chunk, run VAD on each of those chunks, remove the silence portion if it's below min silenece threshold, and transcribe them? 

Or is there something similar to WhisperX being performed?

https://github.com/m-bain/whisperX

Where they merge smaller chunks before transcribing them.

Thanks.

**Comment by trungkienbkhn** [comment]
@ngcheeyuan , hello. You can enable Silero VAD by setting the option `vad_filter=True`. This will be handled in the [logic](https://github.com/SYSTRAN/faster-whisper/blob/master/faster_whisper/transcribe.py#L333).
